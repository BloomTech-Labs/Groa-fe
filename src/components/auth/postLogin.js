import React, {useState} from 'react';
import axios from 'axios';

export default function PostLogin() {



const [input, setInput] = useState({file:''})
    const [ratings, setRatings] = useState({})
    const [loading, setLoading] = useState()

    const handleChange = e => { 
        let data = new FormData()
        data.append('movies', e.target.files[0] , e.target.files[0].name)
        //change user/1/ to be :id number
        //Groabe-env.v3umry9g8h.us-east-1.elasticbeanstalk.com/
        axios.post('http://localhost:4000/api/users/1/upload', data,{
            headers:{
                'Content-Type':'multipart/form-data'
            }
        })
        .then(res => { 
            console.log(res);
            setRatings(res.data)
        }).catch(err => { 
            console.log(err)
        })
        data = new FormData()
        setTimeout(() => {
            setLoading(<h2 className = 'loading' style={{paddingLeft:'5px'}}>.loading</h2>)
        }, 1000);
        setTimeout(() => {
            setLoading(<h2 className = 'loading' style={{paddingLeft:'10px'}}>..loading</h2>)
        }, 2000);
        setTimeout(() => {
            setLoading(<h2 className = 'loading' style={{paddingLeft:'15px'}}>...loading</h2>)
        }, 3000);
        setTimeout(() => {
            setLoading(<h2 className = 'loading' style={{paddingLeft:'20px'}}>....loading</h2>)
        }, 4000);
        // setTimeout(() => {
        //     setLoading('..loading')
        // }, 4000);
        // setTimeout(() => {
        //     setLoading('.loading')
        // }, 5000);
   
}

return (
<div className='PostLoginPage'>
    <div className='container'>
            <h1>Welcome to Groa</h1>

            <h2>As you give us more data, the richer the experience we can deliver.</h2>
            <h2>Show, don't just tell</h2>
        <div className='boxContainer'> 
    <div className='boxLeft'>
        <form className='boxInside'>
        
        <h3>Upload your letterboxd csv file here.</h3>
        <div className='holder'>
            <input className='movie-input'
                type='file'
                placeholder='letterbox csv file here'
                name = 'movies'
                value={input.movieName}  // im not sure why this works but it does.
                onChange={handleChange}
            /> 
        </div>
        {/* <button id="submit-button">Groa</button> */}
        </form>
 {/* END FORM */}        
    </div> 
 {/* END BOX LEFT */}   
    <div className='boxRight'>
        <div className='boxInside'>
        
        <h3>If you dont have a Letterboxd Account take our quiz, so we can get a
        recommenation of movies for you.</h3>
        
        <div className='holder'>
        <button className='btnQuiz'>Take Quiz</button>
        </div>
        
        </div>
    </div>
{/* END BOX RIGHT */}
    </div>
{/* END BOX CONTAINER */}
    </div>
{/* END CONTAINER */}
</div>
   )
 

}
