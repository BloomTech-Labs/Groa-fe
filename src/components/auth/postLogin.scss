.PostLoginPage{
    display: flex;
    justify-content: center;
    border: 2px solid blue;
}

.textholder1{
    
   justify-content: center;
   border: 2px solid green;
   
        h1{
            display: flex;
            justify-content: center;
            font-size: 350%;
        }

        h2{
            display: flex;
            justify-content: center;
        }z

        h3{
            display: flex;
            justify-content: center;
        }
}

.image1{
    display: flex;
    justify-content: center;
    height: 100px;
    width: 200px;
    background: gray;
    margin-bottom: 8%;
}

.image2{
    height: 100px;
    width: 200px;
    background: gray;
}


.container{
    display: flex;
    flex-direction: column; 
    width: 90%;
    border: 2px solid red;
}

.boxContainer{
    display: flex;
    justify-content: center;
    margin-top: 5%;
   border: 2px solid blue;

    .boxLeft{
        width:  285px;
        height: 500px;
        margin-right: 20%;
        border: 2px solid black;

       .boxInside{
            display: flex;
            justify-content: center;
            flex-direction: column;
            padding: 25px;   
        }

        .holder{
            display: flex;
            justify-content: center;
        }   
    }
    
    .boxRight{
        width:  285px;
        height: 400px;
        // margin-left: 40%;
        border: 2px solid black;

        .boxInside{
            display: flex;
            flex-direction: column;
            padding: 25px;

            .holder{
                display: flex;
                justify-content: center;
            }
        }
    }

}

.movie-input{     
    padding: 10px;
    background: gray;
    width: 300px;
}

.btn{
    color: white;
    font-size: 25px;
    background: black;
    height: 51px;
    width: 255px;
}