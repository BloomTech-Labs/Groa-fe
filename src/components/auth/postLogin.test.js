import React from "react";
import { render } from "@testing-library/react";
import postLogin from './postLogin';



describe('Button postlogin', () => {
  it('postlogin be defined', () => {
    expect(postLogin).toBeDefined();
  });

  it('should render postlogin correctly', () => {
    const tree = (
      <postLogin name='post login button test' />
    );
    expect(tree).toMatchSnapshot();
  });
 });