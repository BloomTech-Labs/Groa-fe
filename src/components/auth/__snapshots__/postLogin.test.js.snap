// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button postlogin should render postlogin correctly 1`] = `
<postLogin
  name="post login button test"
/>
`;
